### DISCUSSION TITLE
Modular architecture with Quasar

### DISCUSSION THREAD

**Discussion by Isense-Development** [quasarframework/quasar#11594] [CLI - PWA mode] [answered]
Hi, 

I would like to know if it is possible to build a Quasar application with a modular approach?
I have tried several things but the issue is i cannot access the allready generated router to add routes on the fly based on the modules i create, but it generates the error below because i don't know how to access the router.
Also for the store i face the same issues...

![2021-12-08 14_39_20-Window](https://user-images.githubusercontent.com/56556048/145218040-955bdfef-93a4-48b6-b9db-6d90f4bedb22.png)

The routes should be added to the allready created router in the router/index.js file

![2021-12-08 14_36_47-package json – cart_router js](https://user-images.githubusercontent.com/56556048/145217735-44a34acc-8115-4fc2-b151-b52616ea3013.png)

I have created a register modules js file:

![2021-12-08 14_37_13-package json – register-modules js](https://user-images.githubusercontent.com/56556048/145217738-35b6af17-c9b0-45e1-8d1d-a163f5d76ee1.png)

In my main.js i want to register all the modules:

![2021-12-08 14_37_32-package json – main js](https://user-images.githubusercontent.com/56556048/145217742-1edbce9d-cc0d-4e36-9d06-97fee4252c41.png)

**Comment by metalsadman** [comment]
https://quasar.dev/quasar-cli/boot-files#anatomy-of-a-boot-file
https://forum.quasar-framework.org/topic/3960/access-router-outside-vue/8

**Comment by pdanpdan** [comment, marked as answer]
This does not work in anything, not only in quasar.
You have a child route with path / which is root

**Reply by Isense-Development** [reply]
When i posted it i saw that the "/" was the issue.

**Comment by pdanpdan** [comment]
And then also check if you have a router-view in Module
